packages/cli/src/commands/build/index.ts

```diff
@@ -826,6 +832,43 @@ function mergeFlags(
   });
 }
 
+/**
+ * Takes the build output and writes all the variants into the `variants.json`
+ * file. It'll skip variants that already exist.
+ */
+async function mergeVariants(
+  { output }: Client,
+  buildResults: Iterable<BuildResult | BuildOutputConfig>,
+  outputDir: string
+): Promise<void> {
+  const variantsFilePath = join(outputDir, 'variants.json');
+
+  const variants = (await fs.readJSON(variantsFilePath).catch(error => {
+    if (error.code === 'ENOENT') {
+      return { defintions: {} };
+    }
+
+    throw error;
+  })) as { definitions: Record<string, VariantDefinition> };
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** simply type usage

In this or a follow-up PR: Can we remove the type assertion? If not, can we simplify this complex expression into a couple separate statements?‏

Reply:
The types will come from a different package eventually, so yup, will make sure to update those once it's ready